packages/integration-tests/src/EntityManager.reactiveRules.test.tsx

```diff
@@ -178,16 +178,44 @@ describe("EntityManager.reactiveRules", () => {
     expect(a.numberOfBooks.get).toBe(0);
   });
 
+  it.withCtx("updates derived fields that are foreign keys", async ({ em }) => {
+    // Given an entity with an async derived field
+    const a = newAuthor(em);
+    const b = newBook(em, { author: a, reviews: [{ rating: 10 }] });
+    await em.flush();
+
+    // For some reason .toMatchEntity(b) fails here
+    // expect(a.favoriteBook.get).toMatchEntity(b);
+    expect(a.favoriteBook.get).toBe(b);
```

Comment:
Suggestion: besides asserting against the entity getter, can you use the select helper to assert again the fk in the data base as well?

Oh! I see you added an assertion for that later in the test; can we do a db-based assertion after each em.flush? Just because this is for a derived value.

Reply:
Ok, I moved the `select` assert before the `.get` assert.